sqlglot/generator.py

```diff
@@ -455,19 +460,22 @@ def uniquecolumnconstraint_sql(self, _) -> str:
 
     def create_sql(self, expression: exp.Create) -> str:
         kind = self.sql(expression, "kind").upper()
-        has_before_properties = expression.args.get("properties")
-        has_before_properties = (
-            has_before_properties.args.get("before") if has_before_properties else None
-        )
-        if kind == "TABLE" and has_before_properties:
+        properties = expression.args.get("properties")
+        properties_locs = self.locate_properties(properties) if properties else {}
+        if properties and properties_locs["root_with"]:
+            # subset expression properties to just root and with properties
+            expression.set("properties", exp.Properties(expressions=properties_locs["root_with"]))
```

Comment:
This overwrites the expressions properties, removing the unsupported properties.

If the input isn't re-parsed between tests, then LocationProperty is being removed during the Presto test and not being replaced afterward.

Reply:
I did it this way so I could leave all the code that handles the WITH and ROOT properties unmodified.

Should I take another approach that doesn't modify the original expression?

Reply:
Yes it's not correct to mutate the original expression in generator methods. If you must change the expression, then you need to copy it first.